controllers/gatekeeper_controller.go

```diff
@@ -34,14 +63,56 @@ type GatekeeperReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+// Gatekeeper Operator RBAC permissions to manager Gatekeeper custom resource
 // +kubebuilder:rbac:groups=operator.gatekeeper.sh,namespace="system",resources=gatekeepers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=operator.gatekeeper.sh,namespace="system",resources=gatekeepers/status,verbs=get;update;patch
 
+// Gatekeeper Operator RBAC permissions to deploy Gatekeeper. Many of these
+// RBAC permissions are needed because the operator must have the permissions
+// to grant Gatekeeper its required RBAC permissions.
+
+// Cluster Scoped
+// +kubebuilder:rbac:groups=*,resources=*,verbs=get;list;watch
+// +kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=config.gatekeeper.sh,resources=configs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=config.gatekeeper.sh,resources=configs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=constraints.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=policy,resources=podsecuritypolicies,verbs=use
+// +kubebuilder:rbac:groups=status.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=templates.gatekeeper.sh,resources=constrainttemplates,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=templates.gatekeeper.sh,resources=constrainttemplates/finalizers,verbs=get;update;patch;delete
+// +kubebuilder:rbac:groups=templates.gatekeeper.sh,resources=constrainttemplates/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;clusterrolebindings,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingwebhookconfigurations,verbs=get;list;watch;create;update;patch;delete
+
+// Namespace Scoped
+// +kubebuilder:rbac:groups=core,namespace="system",resources=secrets;serviceaccounts;services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,namespace="system",resources=roles;rolebindings,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,namespace="system",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+
 func (r *GatekeeperReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("gatekeeper", req.NamespacedName)
+	ctx := context.Background()
+	logger := r.Log.WithValues("gatekeeper", req.NamespacedName)
+	logger.Info("Reconciling Gatekeeper")
+
+	if req.Name != defaultGatekeeperCrName {
+		err := fmt.Errorf("Gatekeeper resource name must be '%s'", defaultGatekeeperCrName)
+		logger.Error(err, "Invalid Gatekeeper resource name")
+		// Return success to avoid requeue
+		return ctrl.Result{}, nil
```

Comment:
This will show up in the logs, and that's fine. However, perhaps it would be more user-friendly to output an event using an event recorder. That way it would be easier to know what happened here.

Reply:
Agreed, there's definitely lots of room for improvement. I've filed https://github.com/font/gatekeeper-operator/issues/27 to track it.